ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml

```diff
@@ -1185,6 +1185,106 @@ keys:
+        type: dict
+        keys:
+          bgp:
+            type: dict
+            keys:
+              missing_policy:
+                type: dict
+                keys:
+                  direction_in_action:
+                    type: str
+                    valid_values:
+                      - "deny"
+                      - "deny-in-out"
+                      - "permit"
+                  direction_out_action:
+                    type: str
+                    valid_values:
+                      - "deny"
+                      - "deny-in-out"
+                      - "permit"
+          peer_groups:
+            type: list
+            primary_key: name
+            convert_types:
+              - dict
+            items:
+              type: dict
+              keys:
+                name:
+                  type: str
+                  description: Peer-group name
+                activate:
+                  type: bool
+                missing_policy:
+                  type: dict
+                  keys:
+                    direction_in_action:
+                      type: str
+                      valid_values:
+                        - "deny"
+                        - "deny-in-out"
+                        - "permit"
+                    direction_out_action:
+                      type: str
+                      valid_values:
+                        - "deny"
+                        - "deny-in-out"
+                        - "permit"
+          neighbors:
+            type: list
+            primary_key: ip_address
+            convert_types:
+              - dict
+            items:
+              type: dict
+              keys:
+                ip_address:
+                  type: str
+                activate:
+                  type: bool
+                missing_policy:
+                  type: dict
+                  keys:
+                    direction_in_action:
+                      type: str
+                      valid_values:
+                        - "deny"
+                        - "deny-in-out"
+                        - "permit"
+                    direction_out_action:
+                      type: str
+                      valid_values:
+                        - "deny"
+                        - "deny-in-out"
+                        - "permit"
+          path_selection:
+            type: dict
+            keys:
+              roles:
+                type: dict
+                description: |
+                  Three roles are available as defined in Link-State RFC
+                  * producer
+                  * consumer
+                  * propagator
+
+                  When setting 'producer' to 'True' the CLI
+                  'path-selection' is rendered
+
+                  When setting 'consumer' and / or 'propagator' to 'True'
+                  the followin CLI is rendered
+                  'path-selection role ( propagator | ( consumer [ propagator ] ) )'
```

Comment:
Let's have a call - this is not really logic just how it works.

to configure producer + consumer + propragator you have to do:

```
address-family link-state
  path-selection
  path-selection consumer propagator
```

Reply:
happy to discuss the model I originally had 

```
+        path_selection:
+          enable: <bool>
+          roles:
+            consumer: <bool>
+            propagator: <bool>
```

But `enable` is false - because really without 'role' this enables producer role.

Reply:
I like your original model more, since it would be recognizable when looking at the EOS config.

Reply:
Maybe even
```yaml
path_selection:
  enable: <bool>
  consumer: <bool>
  propagator: <bool>
```

Reply:
except `enable` is wrong and misleading

This is acceptable and valid config:
```
address-family link-state
  path-selection consumer propagator
```

And this
```
address-family link-state
  path-selection
```
actually means "role producer"